File: stacking_cards.py
def min_steps_to_sort_cards(cards, n):
    list_of_max = [1] * n

    for i in range(1, n):
        for j in range(i):
            if cards[i] > cards[j]:
                list_of_max[i] = max(list_of_max[i], list_of_max[j] + 1)

    min_steps = n - max(list_of_max)
    
    return min_steps

def main():

    data_sets = int(input())
    results = [0]*data_sets
    for i in range(data_sets):
        n = int(input())
        cards = list(map(int, input().split()))
        results[i] = min_steps_to_sort_cards(cards, n)

    for result in results: print(result)

main()

File: stacking_cards.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int min_steps_to_sort_cards(vector<int>& cards, int n) {
    vector<int> card_positions(n + 1, 0);
    vector<int> lis(n, 1);

    for (int i = 1; i < n; ++i) {
        for (int j = 0; j < i; ++j) {
            if (cards[i] > cards[j]) {
                lis[i] = max(lis[i], lis[j] + 1);
            }
        }
    }

    int min_steps = n - *max_element(lis.begin(), lis.end());
    return min_steps;
}

int main() {
    int data_sets;
    cin >> data_sets;
    vector<int> results(data_sets);

    for (int i = 0; i < data_sets; ++i) {
        int n;
        cin >> n;
        vector<int> cards(n);
        for (int j = 0; j < n; ++j) {
            cin >> cards[j];
        }
        results[i] = min_steps_to_sort_cards(cards, n);
    }

    for (int result : results) {
        cout << result << endl;
    }

    return 0;
}
